lib/common/huf.h

```diff
@@ -213,6 +213,14 @@ size_t HUF_compress4X_repeat(void* dst, size_t dstSize,
                        void* workSpace, size_t wkspSize,    /**< `workSpace` must be aligned on 4-bytes boundaries, `wkspSize` must be >= HUF_WORKSPACE_SIZE */
                        HUF_CElt* hufTable, HUF_repeat* repeat, int preferRepeat, int bmi2);
 
+/* Incorporate fast check(s) on the suspicion that this data is not compressible and back off
+   quickly to raw format if so. */
+size_t HUF_compress4X_repeat_fastCheck(void* dst, size_t dstSize,
+                       const void* src, size_t srcSize,
+                       unsigned maxSymbolValue, unsigned tableLog,
+                       void* workSpace, size_t wkspSize,   /**< `workSpace` must be aligned on 4-bytes boundaries, `wkspSize` must be >= HUF_WORKSPACE_SIZE */
+                       HUF_CElt* hufTable, HUF_repeat* repeat, int preferRepeat, int bmi2, unsigned suspectUncompressible);
```

Comment:
minor : api design : 

as we get more and more parameters in the function signature,
it becomes more and more difficult to understand what this `1` or this `0` means.
For clarity, we end up commenting them, so that the reader can track what each value means.
`... , 1 /* suspected uncompressible */);`.

Another possibility is to use `enum`.
`..., HUF_suspectedUncompressible);`
In such a context, it's essentially a compiler-checked comment.

Reply:
After some thought, added comments rather than an enum; this is really just a stand-in for a boolean value so I think the argument name is sufficient to convey the meaning as far as the function definitions are concerned..